app/scripts/detect-multiple-instances.test.js

```diff
@@ -0,0 +1,58 @@
+import { strict as assert } from 'assert';
+import browser from 'webextension-polyfill';
+import sinon from 'sinon';
+import {
+  PLATFORM_CHROME,
+  METAMASK_BETA_CHROME_ID,
+  METAMASK_PROD_CHROME_ID,
+  METAMASK_FLASK_CHROME_ID,
+} from '../../shared/constants/app';
+import {
+  checkForMultipleVersionsRunning,
+  onMessageReceived,
+} from './detect-multiple-instances';
+import * as util from './lib/util';
+
+describe('multiple instances running detector', function () {
+  const PING_MESSAGE = 'isRunning';
+
+  it('should send ping message to multiple instances', async function () {
+    const sandbox = sinon.createSandbox();
```

Comment:
Generally we shouldn't use sandboxes. Sinon sandboxes are meant for cases where we want to manage multiple separate sets of mocks independently from each other. In all other cases, the default sandbox is easier and safer to use.

You should also call `sinon.restore()` in the `beforeEach` hook, to ensure that all stubs are properly cleaned up after each test. In addition to helping keep tests isolated, this is recommend in the Sinon general setup docs as a way to prevent memory leaks.

Reply:
Good point!

Reply:
It looks like you're still using a sandbox. The "default sandbox" refers to using the global `sinon` instance directly. No use of `createSandbox` at all.

Reply:
Done! Using sanbox now only for `metamask-controller.test.js` – where it's already used

Reply:
Perfect, thanks!